public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistory.tsx

```diff
@@ -0,0 +1,368 @@
+import { css, cx } from '@emotion/css';
+import React, { useCallback, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useMeasure } from 'react-use';
+
+import { GrafanaTheme2, TimeRange } from '@grafana/data';
+import { isFetchError } from '@grafana/runtime';
+import { SceneComponentProps, SceneObjectBase, sceneGraph } from '@grafana/scenes';
+import {
+  Alert,
+  Button,
+  Field,
+  Icon,
+  Input,
+  Label,
+  LoadingBar,
+  Stack,
+  Text,
+  Tooltip,
+  useStyles2,
+  withErrorBoundary,
+} from '@grafana/ui';
+import { EntityNotFound } from 'app/core/components/PageNotFound/EntityNotFound';
+import { Trans, t } from 'app/core/internationalization';
+import {
+  GrafanaAlertStateWithReason,
+  getReasonFromState,
+  isAlertStateWithReason,
+  isGrafanaAlertState,
+  mapStateWithReasonToBaseState,
+} from 'app/types/unified-alerting-dto';
+
+import { stateHistoryApi } from '../../../api/stateHistoryApi';
+import { GRAFANA_DATASOURCE_NAME } from '../../../utils/datasource';
+import { stringifyErrorLike } from '../../../utils/misc';
+import { AlertLabels } from '../../AlertLabels';
+import { CollapseToggle } from '../../CollapseToggle';
+import { STATE_HISTORY_POLLING_INTERVAL } from '../state-history/LokiStateHistory';
+import { LogRecord } from '../state-history/common';
+import { useRuleHistoryRecords } from '../state-history/useRuleHistoryRecords';
+
+const LIMIT_LABELS = 5;
+const LIMIT_EVENTS = 250;
+
+const HistoryEventsList = ({ timeRange }: { timeRange?: TimeRange }) => {
+  const styles = useStyles2(getStyles);
+  const { useGetRuleHistoryQuery } = stateHistoryApi;
```

Comment:
I was using the same notation we use in [the existing code in alert state history](https://github.com/grafana/grafana/blob/8d36949f61c0158753029f6fb84c7548b2572b46/public/app/features/alerting/unified/components/rules/state-history/LokiStateHistory.tsx#L32) , is there any reason why you prefer this notation?

Reply:
Yep, using the `endpoints` notation it will correctly (at least in my editor) allow me to jump to the definition of that endpoint, I can't seem to make that work with de-structuring the hooks. Maybe it's a local setup problem though.

Reply:
good point! 💯